src/components/molecules/ModalDialog.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Modal from '@mui/material/Modal';
+import styled from 'styled-components';
+import CircularButton from '../atoms/buttons/CircularButton';
+import Button from '@mui/material/Button';
+import { handleBuyOrder } from "../../services/brokerService";
+
+interface Props {
+    isOpen: boolean;
+    handleClose: () => void;
+    ticker: string;
+    price: number;
+    name: string;
+}
+
+const ContentContainer = styled.div`
+    display: flex;
+`
+const InfoContainer = styled.div`
+    width: 50%;
+`
+
+const GraphContainer = styled.div`
+    color: white;
+    display: flex;
+    align-items: center;
+    justify-content: end;
+    padding: 15px 0 15px 0;
+    width: 50%;
+` 
+const Graph = styled.div`
+    width: 50vh;
+    height: 50vh;
+    background-color: #3F433A;
+    color: white;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+` 
+
+const CancelButtonContainer = styled.div`
+    display: flex;
+    justify-content: flex-end;
+    align-items: center;
+    width: 100%;
+`
+
+const style = {
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: '70%',
+  height: '70%',
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  p: 4,
+};
+
+export default function ModalDialog({isOpen, handleClose, ticker, price, name}: Props) {
+
+  return (
+    <div>
+      <Modal
+        open={isOpen}
+        onClose={handleClose}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>
+            <CancelButtonContainer>
+                <CircularButton handleClick={handleClose}>
+                    <>X</>
+                </CircularButton>
+            </CancelButtonContainer>
+        <ContentContainer>       
+            <InfoContainer>
+                <Typography id="modal-modal-title" variant="h6" component="h2">
```

Comment:
If we use Typography, I would rather have it reused via an atom. If we minimise its implementation, it will be easier to change UI lib.